test_runner/src/main/kotlin/ftl/run/platform/RunAndroidTests.kt

```diff
@@ -38,9 +39,15 @@ internal suspend fun runAndroidTests(args: AndroidArgs): TestResult = coroutineS
 
     args.resolveApks().forEach { apks: ResolvedApks ->
         val testShards = apks.test?.let { test ->
-            AndroidTestShard.getTestShardChunks(args, test).also {
-                allTestShardChunks += it
+            AndroidTestShard.getTestShardChunks(args, test)
+        }
+        // We can't return if testShards is null since it can be a robo test.
+        if (testShards != null) {
```

Comment:
Thanks for implementing logic to skip empty apks!

If this is not a problem for you, could you consider rewriting this with (example) `testShards?.let {...}`
Regardless, I like the solution

Reply:
Absolutely, refactoring it!